uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -574,6 +563,28 @@ define([
       }
     }
 
+    /**
+     * Initialises time for a given date type.
+     * In general cases simply extracts the time from the date string and
+     * sets the time to the correspondent times property.
+     * If the time is outside the allowed range (for example after work parrern change),
+     * then it sets the minimum allowed time for "from" time
+     * and the maximum allowed time for "to" time.
+     *
+     * @param {String} dateType from|to
+     */
+    function initTime (dateType) {
+      var time = moment(vm.request[dateType + '_date']).format('HH:mm');
+      var timeObject = vm.uiOptions.times[dateType];
+
+      if (moment.duration(time).subtract(moment.duration(timeObject.min)).asHours() <= 0 ||
```

Comment:
can we name these conditions?

```js
var isBeforeWorkPatternRange = moment.duration(time).subtract(moment.duration(timeObject.min)).asHours() <= 0;
var isAfterWorkPatternRange = moment.duration(time).subtract(moment.duration(timeObject.max)).asHours() >= 0;

if (isBeforeWorkPatternRange || isAfterWorkPatternRange) {
  // ...
}
```

and, why not use `getTimeDifferenceInHours` ?

Reply:
Why isn't this condition like this instead?

```js
function initTime (dateType) {
  var isBeforeWorkPatternRange, isAfterWorkPatternRange;
  var time = moment(vm.request[dateType + '_date']).format('HH:mm');
  var timeObject = vm.uiOptions.times[dateType];
  var isBeforeWorkPatternRange = getTimeDifferenceInHours(timeObject.min) <= 0;
  var isAfterWorkPatternRange = getTimeDifferenceInHours(timeObject.min) >= 0;
  
  if (dateType === 'from' && isBeforeWorkPatternRange) {
    return timeObject.min;
  } else if (dateType === 'to' && isAfterWorkPatternRange) {
    return timeObject.max;
  }
}
```